modules/ex/src/clj/exoscale/ex.clj

```diff
@@ -298,3 +301,41 @@
    (when-not (s/valid? spec x)
      (throw (ex-invalid-spec spec x data)))
    x))
+
+(s/def ::ex-map
+  (s/keys :req [::message
+                ::type
+                ::data]
+          :opt [::deriving
+                ::cause]))
+(extend-protocol p/Datafiable
+  clojure.lang.ExceptionInfo
+  (datafy [x]
+    (let [{:keys [type] :as data} (ex-data x)]
+      (if type
+        (let [cause (ex-cause x)
+              deriving (parents type)]
+          (cond-> {::type type
+                   ::message (ex-message x)
+                   ::data data}
+            (seq deriving)
+            (assoc ::deriving deriving)
+            (some? cause)
+            (assoc ::cause (p/datafy cause))))
+        (Throwable->map x)))))
+
+(s/fdef map->ex-info
+  :args (s/cat :ex-map ::ex-map
+               :options (s/? (s/keys :opt [::derive?]))))
+(s/def ::derive? boolean?)
+(defn map->ex-info
```

Comment:
Where would you see being used? Going from an error response back to a throwable when communicating between services?

Reply:
Exactly. We have this between an internal service and its client used in another service that basically just interprets/retrows errors. I can give your more details on IRC.

Reply:
Excelllent idea